### PR TITLE
Elimate `requires_grad=True` warnings from device tests

### PR DESCRIPTION
**Pull request by dime10**
Small PR to remove `requires_grad=True` warnings from the device tests in:

- tests/devices
- pennylane/devices/tests

These warnings were introduced in anticipation of Pennylane v0.20.0 making the default `requires_grad=False` for variational parameters.

### CODE REVIEW

**Comment by dime10** [comment]
[sc-11205]

**Review by albi3ro — COMMENTED** [review]
Thanks for this :)

Just needs a changelog entry and I'll approve.

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -280,6 +280,10 @@
 * The `requires_grad` attribute of `qml.numpy.tensor` objects is now
   preserved when pickling/unpickling the object.
   [(#1856)](https://github.com/PennyLaneAI/pennylane/pull/1856)
+  
+* Device tests no longer throw warnings about the `requires_grad`
+  attriubte of variational parameters.
```
Comment (albi3ro) [inline doc/releases/changelog-dev.md:285]:
```suggestion
  attribute of variational parameters.
```

**Review by albi3ro — APPROVED** [review]
:)  Just the one spelling thing.

File: doc/releases/changelog-dev.md (hunk 1 above)
Comment (dime10) [reply doc/releases/changelog-dev.md:285]: Thanks for catching that :)

**Pull request merged**